index.ts

```diff
@@ -258,6 +256,56 @@ export class Project {
       dep.baseDir = path.join(this.baseDir, 'node_modules', dep.name);
       dep.writeSync();
     }
+    for (let [name, { dir: target }] of this.dependencyLinks) {
+      this.writeLinkedPackage(name, target);
+    }
+  }
+
+  private writeLinkedPackage(name: string, target: string) {
+    let targetPkg = require(path.join(target, 'package.json'));
+    let peers = new Set(Object.keys(targetPkg.peerDependencies ?? {}));
+    let destination = path.join(this.baseDir, 'node_modules', name);
+
+    if (peers.size === 0) {
+      // no peerDeps, so we can just symlink the whole package
+      fs.ensureSymlinkSync(target, destination, 'dir');
+      return;
+    }
+
+    // need to reproduce the package structure in our own location
+    this.hardLinkContents(target, destination);
+
+    for (let section of ['dependencies', 'peerDependencies']) {
+      if (targetPkg[section]) {
+        for (let depName of Object.keys(targetPkg[section])) {
+          if (peers.has(depName)) {
+            continue;
+          }
+          let depTarget = resolvePackagePath(depName, target);
```

Comment:
The caching provided by default by `resolvePackagePath` could result in some unexpected behavior. Currently it mimics `require` cache semantics (cache persistent for the duration of the process/context). But it may be possible to accidental pollute this cache, especially when working through a serious of different test scenarios in the same process.

To avoid this, but still benefit from the consistency and speed-up the cache offers,  we may want to consider [a unique cache](https://github.com/stefanpenner/resolve-package-path#provide-an-alternative-cache) per top-level fixturify-project? This then enforces a pretty reasonable "deps cant change" for any given project, but between projects they are free-game to change.

WDYT?

Reply:
Added a unique cache per suggestion.